### PR TITLE
WIP. pj-18-create-post, do w/o tests

### PR DESCRIPTION
**Pull request by NykoEugen**

### CODE REVIEW

File: posts/decorator.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.shortcuts import redirect
+
+
+def user_required(view_func):
```
Comment (acman) [inline posts/decorator.py]: it is implemented in django, so there is no reason to write custom one

File: posts/views.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from django.utils.decorators import method_decorator
+from django.views import View
+
+from categories.models import Category
+from posts.decorator import user_required
+from posts.forms import PostForm
+
+
+@method_decorator(user_required, name="dispatch")
+class CreatePostView(View):
```
Comment (acman) [inline posts/views.py]: method decorator works only for methods
there is a good place to use `LoginRequiredMixin`, I think it has everything u need
```suggestion
class CreatePostView(LoginRequiredMixin, View):
```

**Review by acman — CHANGES_REQUESTED** [review]
Overall looks good! User logined check should be updated

File: posts/tests.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            name="Test Category",
+            main_category=self.main_category,
+        )
+
+        self.create_view_url_get = reverse(
+            "posts:create", kwargs={"category_id": self.category.pk}
+        )
+        self.create_view_url_post = reverse(
+            "posts:create", kwargs={"category_id": self.category.pk}
+        )
```
Comment (acman) [inline posts/tests.py]: suppose first url should get post

Comment (acman) [reply posts/tests.py]: get - should render PostForm
post - create Post

**Review by acman — CHANGES_REQUESTED** [review]
some invalid tests

File: posts/tests.py (hunk 3 above)
Comment (acman) [reply posts/tests.py]: I see, it is valid but it is not clear that self.create_view_url_get and self.create_view_url_post have the same url. So better to have only one consistent
```suggestion
        self.create_view_url = reverse(
            "posts:create", kwargs={"category_id": self.category.pk}
        )
```

**Review by acman — CHANGES_REQUESTED** [review]

File: posts/tests.py (hunk 3 above)
Comment (NykoEugen) [reply posts/tests.py]: Oh, I see it, but don't pay attention.

**Review by acman — APPROVED** [review]
👍

**Pull request merged**